ci-operator/step-registry/ibmcloud/provision/vpc/ibmcloud-provision-vpc-commands.sh

```diff
@@ -219,7 +219,18 @@ rg_id=$("${IBMCLOUD_CLI}" resource group $resource_group --id)
 ## Create the VPC
 CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
 vpc_name="${CLUSTER_NAME}-vpc"
-declare -a ZONES=("${region}-1" "${region}-2" "${region}-3")
+
+readarray -t ZONES  < <(${IBMCLOUD_CLI} is zones -q | grep ${region} | awk '{print $1}')
+max_zones="${#ZONES[@]}"
+if [ ${ZONES_COUNT} -gt ${max_zones} ]; then
```

Comment:
I do not think we need these two `if` condition, but directly use `ZONES=("${ZONES[@]:0:${ZONES_COUNT}}")`.


Per your code logic, when ZONES_COUNT > max_zones, ZONES_COUNT is reassigned to `max_zones` (e.g: 4), but the following 2nd if condition, the code only respect the hard code - 3 as `max_zones`
`

Reply:
I just consider max_zones  <= 3

Reply:
You already set `ZONES` by ibmcloud command, so that means, the size of `ZONES` is determinate by ibmcloud, we can not just assume the length is for ever "3".